tools/clang/lib/SPIRV/SpirvContext.cpp

```diff
@@ -231,14 +231,16 @@ const HybridPointerType *SpirvContext::getPointerType(QualType pointee,
 
 FunctionType *
 SpirvContext::getFunctionType(const SpirvType *ret,
-                              llvm::ArrayRef<const SpirvType *> param) {
+                              llvm::ArrayRef<const SpirvType *> param,
+                              bool isMember) {
```

Comment:
you should also modify line 237 to
```cpp
FunctionType type(ret, param, isMember);
```

Reply:
Yes, I think it is working. it is not doing a pointer comparison for finding the object. We have overridden the comparison function. see `isEqual` function in FunctionTypeMapInfo:

https://github.com/microsoft/DirectXShaderCompiler/blob/b0731e7f2d45b63db4c80907bcfd4cab2e3c666c/tools/clang/include/clang/SPIRV/SpirvContext.h#L107-L110

Comparison returns true if the pointers are the same (`LHS == RHS `) or if the objects are equal (`*LHS == *RHS`)